rust-runtime/aws-smithy-types/src/date_time/mod.rs

```diff
@@ -47,7 +47,7 @@ const NANOS_PER_SECOND_U32: u32 = 1_000_000_000;
 /// The [`aws-smithy-types-convert`](https://crates.io/crates/aws-smithy-types-convert) crate
 /// can be used for conversions to/from other libraries, such as
 /// [`time`](https://crates.io/crates/time) or [`chrono`](https://crates.io/crates/chrono).
-#[derive(Debug, PartialEq, Clone, Copy)]
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
```

Comment:
If I am not mistaken (@unexge can comment on this), we need to derive the same also on the Python wrappers that are found inside `aws-smithy-http-server-python/src/types.rs`.